src/overlays/actors/ovl_En_Bb/z_en_bb.c

```diff
@@ -36,118 +47,667 @@ const ActorInit En_Bb_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bb/func_808C2B1C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bb/func_808C2B94.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bb/func_808C2BD0.s")
+/**
+ * This maps a given limb based on its limbIndex to its appropriate index
+ * in the bodyPartsPos/Velocity arrays. An index of -1 indicates that the
+ * limb is not part of the bodyParts arrays.
+ */
+static s8 sLimbIndexToBodyPartsIndex[] = {
+    -1, -1, -1, -1, 0, -1, -1, -1, 1, -1, -1, -1, -1, 2, -1, 3,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bb/func_808C2C38.s")
+/**
+ * The last element of the bodyParts arrays is a duplicate of the cranium
+ * limb, which is then offset by a certain amount. There is no display list
+ * associated with this, so it is only used for effects.
+ */
+static Vec3f sDuplicateCraniumBodyPartOffset = { 1000.0f, -700.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bb/func_808C2CB4.s")
+void EnBb_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnBb* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bb/func_808C2CF0.s")
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &gBubbleSkel, &gBubbleFlyingAnim, this->jointTable, this->morphTable,
+                   BUBBLE_LIMB_MAX);
+    Collider_InitAndSetSphere(globalCtx, &this->collider, &this->actor, &sSphereInit);
+    ActorShape_Init(&this->actor.shape, 1500.0f, ActorShadow_DrawCircle, 35.0f);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bb/func_808C2D78.s")
+    this->flameScaleY = 0.8f;
+    this->flameScaleX = 1.0f;
+    this->actor.world.pos.y += 50.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bb/func_808C2E34.s")
+    if (EN_BB_ATTACK_RANGE(&this->actor) == 0xFF) {
+        this->attackRange = 200.0f;
+    } else {
+        this->attackRange = EN_BB_ATTACK_RANGE(&this->actor) * 4.0f;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bb/EnBb_Update.s")
+    EnBb_SetupFlyIdle(this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bb/func_808C32EC.s")
+void EnBb_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnBb* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bb/func_808C3324.s")
+    Collider_DestroySphere(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bb/EnBb_Draw.s")
+/**
+ * Checks to see if the Bubble is touching a wall. If it is, and if the
+ * Bubble is facing directly "into" the wall, then rotate it away from
+ * the wall.
+ */
+void EnBb_CheckForWall(EnBb* this) {
+    s16 yawDiff;
+
+    if (this->actor.bgCheckFlags & 8) {
+        yawDiff = this->actor.shape.rot.y - this->actor.wallYaw;
+        if (ABS_ALT(yawDiff) >= 0x4001) {
```

Comment:
```suggestion
        if (ABS_ALT(yawDiff) > 0x4000) {
```
The rounder number is probably preferred.